src/scalar_store.rs

```diff
@@ -97,55 +102,115 @@ impl<'a, F: PrimeField> ScalarStore<F> {
     pub fn finalize(&mut self, store: &Store<F>) {
         self.add_pending_scalar_ptrs(store);
     }
+    pub fn get_expr(&self, ptr: &ScalarPtr<F>) -> Option<&ScalarExpression<F>> {
+        let x = self.scalar_map.get(ptr)?;
+        (*x).as_ref()
+    }
+
+    pub fn get_cont(&self, ptr: &ScalarContPtr<F>) -> Option<&ScalarContinuation<F>> {
+        let x = self.scalar_cont_map.get(ptr)?;
+        (*x).as_ref()
+    }
+    pub fn to_store(&mut self) -> Option<Store<F>> {
+        if self.pending_scalar_ptrs.is_empty() {
+            None
+        } else {
```

Comment:
This condition confuses me. I think `self.pending_scalar_ptrs` *should* be empty if a `ScalarStore` has been fully created. It should only be non-empty as an intermediate state when new `Ptr`s are being added to it.

As written, it looks like this will return `None` under the conditions I would expect it to be called. Presumably that's not what happens in existing tests, so I'm a little unsure what's actually going on.

Reply:
The thinking is that you can only convert a finalized/fully initialized `ScalarStore` back to `Store`. If you have `pending_scalar_ptrs` then you should first finish the conversion from `Store` before you try going back to `Store`